### ISSUE TITLE
Upgrade pillow support to 11.0.0

### ISSUE THREAD

**Issue by bob-jones-cs** [gradio-app/gradio#9818]
- [x] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  
N/A

**Describe the solution you'd like**  
I'd like gradio to be upgraded to support the latest version of pillow (11.0.0).

**Additional context**  
N/A

**Comment by abidlabs** [comment]
Thanks @bob-jones-cs I looked through the breaking changes and we should be okay with all of them. Would you like to open a PR to support pillow 11?

**Comment by pjbaron** [comment]
This would still be a good change although I'm not qualified to take it on myself.

I have encountered multiple recent projects that require pillow 11.0 or pillow 11.1 and gradio 4.43 or 4.44. The requirement is embedded in their installations so I suspect this combination must have worked until some recent change.
In one instance changing either gradio or pillow had critical knock-on problems with the required dlib version and I decided not to knock over the next domino!

e.g. https://github.com/aigc3d/LHM/issues/30#issuecomment-2750454920
